src/utils.ts

```diff
@@ -68,27 +68,34 @@ function isLazyCategoricalArray<
 	K extends UIntType,
 	S extends Readable,
 >(
-	array: LazyCategoricalArray<K, L, S> | zarr.Array<L, S> | SparseArray<N>,
+	array:
+		| LazyCategoricalArray<K, L, S>
+		| zarr.Array<L, S>
+		| SparseArray<N, IndexType, S>,
 ): array is LazyCategoricalArray<K, L, S> {
-	return (array as LazyCategoricalArray<K, L, S>).categories !== undefined;
+	return "categories" in array;
```

Comment:
I assume my simplification is correct and `categories` can’t be present (while set to `undefined`) in the other types?

Reply:
I can't think of a reason it would be